commands/src/main/java/org/wildfly/extras/creaper/commands/ra/AddConnectionFactoryToRA.java

```diff
@@ -0,0 +1,147 @@
+    private String poolName;
+    private String localJndiName;
+    private String resourceAdapterId;
+    private String userName;
+    private String password;
+    private boolean security = false;
+    private boolean tracking = true;
+    private Map<String, String> properties = new HashMap<String, String>();
+
+    private AddConnectionFactoryToRA(Builder builder) {
+        this.poolName = builder.poolName;
+        this.localJndiName = builder.localJndiName;
+        this.resourceAdapterId = builder.resourceAdapterId;
+        this.userName = builder.userName;
+        this.password = builder.password;
+        this.security = builder.security;
+        this.properties = builder.properties;
+        this.tracking = builder.tracking;
+    }
+
+    @Override
+    public void apply(OnlineCommandContext ctx) throws Exception {
+        Operations ops = new Operations(ctx.client);
+        Address address = Address.subsystem("resource-adapters").and("resource-adapter", resourceAdapterId)
+                .and("connection-definitions", poolName);
+        ops.add(address, Values.of("class-name", "org.jboss.resource.adapter.jms.JmsManagedConnectionFactory")
+                .and("jndi-name", localJndiName));
+        if (security) {
+            ops.writeAttribute(address, "security-application", true);
+        }
+        if (userName != null) {
+            ops.writeAttribute(address, "recovery-username", userName);
+        }
+        if (password != null) {
+            ops.writeAttribute(address, "recovery-password", password);
+        }
+        ops.writeAttribute(address, "tracking", tracking);
+        if (properties.size() > 0) {
+
+            Iterator it = properties.entrySet().iterator();
+            while (it.hasNext()) {
+                Map.Entry pair = (Map.Entry) it.next();
+                Address configPropsAddress = Address.subsystem("resource-adapters")
+                        .and("resource-adapter", resourceAdapterId)
+                        .and("connection-definitions", poolName).and("config-properties", (String) pair.getKey());
+                ops.add(configPropsAddress, Values.of("value", (String) pair.getValue()));
+            }
+
+        }
+
+
+    }
+
+    public static final class Builder {
+        private String poolName;
+        private String localJndiName;
+        private String resourceAdapterId;
+        private String userName = null;
+        private String password = null;
+        private boolean security = false;
+        private boolean tracking = true;
+        private Map<String, String> properties = new HashMap<String, String>();
+
+        /**
+         * @param poolName          unique id for CF (not used for jndi lookup)
+         * @param localJndiName     jndi name for CF used for jndi lookup from apps
+         * @param resourceAdapterId id of the RA where this CF will be added
+         */
+        public Builder(String poolName, String localJndiName, String resourceAdapterId) {
```

Comment:
3 constructors parameters of the same type are the absolute maximum. I'm fine with it, but please consider if breaking these down into separate builder methods wouldn't be better.